scripts/data_prep/convert_dataset_json.py

```diff
@@ -161,7 +163,11 @@ def generate_samples(
             if truncate_num_samples is not None and n_samples == truncate_num_samples:
                 return
             n_samples += 1
-            yield {k: v[idx] for k, v in batch.items()}
+            yield {
+                k:
+                v[idx].numpy() if isinstance(v[idx], torch.Tensor) else v[idx]
```

Comment:
where do the tensors come from? in fact I don't think this function is even used anywhere?

Reply:
Oh yeah weird seems it's not even used in this script. But it's used in the HF conversion script -- the dataloader will automatically emit batches of Tensor type due to the default collate function, so we convert them back to numpy arrays here.

Reply:
I thought it was a bit odd that we're using a dataloader to wrap the `ConcatTokensDataset` that's returned from `build_hf_dataset` as well. Is it mainly to get multiple workers going at once, which makes the script faster or something?

Reply:
deleted the `generate_samples` function from `convert_dataset_json.py`

Reply:
yeah i think it was for leveraging torch for multiprocessing and what not. but these scripts were written a long time ago and probably have room for improvement :)